src/tiled/commanddialog.cpp

```diff
@@ -76,17 +83,56 @@ void CommandDialog::setShortcut(const QKeySequence &keySequence)
         mUi->treeView->model()->setShortcut(current, keySequence);
 }
 
-void CommandDialog::updateKeySequenceEdit(const QModelIndex &current, const QModelIndex &)
+void CommandDialog::setSaveBeforeExecute(int state)
+{
+    const QModelIndex &current = mUi->treeView->currentIndex();
+    if (current.row() < mUi->treeView->model()->rowCount(QModelIndex()))
+        mUi->treeView->model()->setSaveBeforeExecute(current, state);
+}
+
+void CommandDialog::setCommand(const QString &text)
+{
+    const QModelIndex &current = mUi->treeView->currentIndex();
+    if (current.row() < mUi->treeView->model()->rowCount(QModelIndex()))
+        mUi->treeView->model()->setCommand(current, text);
+}
+
+void CommandDialog::updateWidgets(const QModelIndex &current, const QModelIndex &)
 {
     if (current.row() < mUi->treeView->model()->rowCount(QModelIndex()) - 1) {
-        mUi->keySequenceEdit->setEnabled(true);
-        mUi->clearButton->setEnabled(true);
+        enableWidgets(true);
+
         mUi->keySequenceEdit->setKeySequence(mUi->treeView->model()->shortcut(current));
-    } else {
+        mUi->saveBox->setChecked(mUi->treeView->model()->saveBeforeExecute(current));
+        mUi->commandEdit->setText(mUi->treeView->model()->command(current));
+    }
+    else
+        enableWidgets(false);
+}
+
+void CommandDialog::openFileDialog()
+{
+    QString caption = tr("Select executable");
+    QString dir = tr("/bin");
+    QString executableName = QFileDialog::getOpenFileName(this, caption, dir);
+
+    if (!executableName.isEmpty())
+        mUi->commandEdit->setText(executableName);
+}
+
+void CommandDialog::enableWidgets(const bool enable)
```

Comment:
Please merge this function into `updateWidgets`. I see no gain in having this as a separate method.